### DISCUSSION TITLE
Course Related

### DISCUSSION THREAD

**Discussion by pk1098** [snigi-gupta/AttendingUniversityAtBuffalo#44] [Q&A] [answered]
Hi,

First of all, you are doing great work, moreover, your reply to my   AI/ML( #2 ) query was very helpful. I had started in the spring semester as a practicum track student and have completed one course(Intro to ML). 
I am planning to enroll in the courses given below
1) Distributed Systems by Bina Ramamurthy
2) Intro to PDP by Jaroslaw Zola
3) Computer Vision and Image Processing by Junsong Yuan
4) Analysis of Algorithms Xin He

Q1-> Is the above-given combination too heavy considering that I have a decent knowledge of Algorithms and Datastructure
Q2-> If yes, can you suggest some easy or medium level courses 
Q3->Would you recommend taking Computer Vision by Junsong Yuan in the Fall because I have read through your course review and you have mentioned that the professor was great. The reviews on rate my professor are not good for Yuan hence asking the question.

**Comment by snigi-gupta** [comment, marked as answer]
Hi @pk1098,

So PDP is heavy, and the rest seem of medium difficulty. I would say this is a heavy course load. It might be difficult to keep up with the projects of all these courses at the same time. You would want to keep some sanity alive during a sem, I would suggest swap one medium course difficult level with an easy course. To more about those combinations [refer to this](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/tree/main/Computer%20Science%20Important%20Tips).

I don't think so you should choose a subject based on just easy/medium/hard. Also, see how that course can help you get a job. You can also swap a course with a seminar and they fairly have a lesser workload. 

Reviews keep changing each year based on the usual perspective of that batch. I would suggest you audit the course. That would be the best way to judge whether or not to take the course. For me his classes were really fun and interesting, but yes, the exams were tough. His course slides have only images and no content/text, so you have to attend each class. He doesn't record it. Some concepts were taught broadly, but the questions were asked in-depth. I got a B (also because I had 3 exams on the same day, which is very unusual for UB). I did not like their TAs though, coz they seemed confused themselves. It is possible there are new TAs. 

I hope this helps, if you have more questions, do ask! :)

**Reply by pk1098** [reply]
Thanks for replying. I was planning to swap the Computer Vision course with Blockchain as I am interested and there is something new that I will learn, but the only problem is that the same course is being offered by Bina Ramamurthy on Coursera so if you have any idea about the difficulty of the course or any senior who has taken the course at UB please let me know.

Thank You!

**Reply by snigi-gupta** [reply]
There is a course review [here ](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/blob/main/Computer%20Science%20Important%20Tips/CourseReview.md). I am not sure if you already read the review. You can get in touch with Sourav who took this course and wrote the review. (his LinkedIn is mentioned on this page) *But please be **very professional** in asking your queries* In case you do not get a response, then don't bother with follow-ups.*
